plugins/destination/managed_writer.go

```diff
@@ -15,39 +15,44 @@ import (
 
 type worker struct {
 	count int
-	wg    *sync.WaitGroup
+	wg    sync.WaitGroup // implies usage by pointer only
+
 	ch    chan schema.CQTypes
 	flush chan chan bool
+
+	table   *schema.Table
+	srcSpec *specs.Source
 }
 
-func (p *Plugin) worker(ctx context.Context, metrics *Metrics, table *schema.Table, ch <-chan schema.CQTypes, flush <-chan chan bool) {
+func (p *Plugin) worker(ctx context.Context, metrics *Metrics, w *worker) {
```

Comment:
Maybe let's call it `runWorker`, because it's already passing a `worker` in as an argument